Client/Frontend/UserContent/UserScripts/PaymentRequest.js

```diff
@@ -17,13 +20,31 @@ class $<paymentreq> {
   constructor (methodData, details) {
       this.methodData = JSON.stringify(methodData)
       this.details = JSON.stringify(details)
+      this.state = $<paymentreq>_CREATED
   }
 
   canMakePayment() {
-    return true;
+    const methodData = JSON.parse(this.methodData)
+    const details = JSON.parse(this.details)
+    const state = this.state
+    return new Promise(
+      function (resolve, reject) {
+        if (state != $<paymentreq>_CREATED) {
+          reject(new DOMException("canMakePayment error", "InvalidStateError"))
+          return
+        }
+        if (!methodData.some(e => e.supportedMethods === 'bat')) {
+          resolve(false)
```

Comment:
Should we reset `this.state` here?

Reply:
The `PaymentRequest` is still active and `PaymentRequest.show` can still be invoked, although it will return an error from the client.